pubsub/azure/eventhubs/eventhubs.go

```diff
@@ -620,6 +630,89 @@ func (aeh *AzureEventHubs) createCheckpointStore() (checkpointStore azeventhubs.
 	return checkpointStore, nil
 }
 
+// Ensures that the container exists in the Azure Storage Account.
+// This is done to preserve backwards-compatibility with Dapr 1.9, as the old checkpoint SDK created them automatically.
+func (aeh *AzureEventHubs) ensureStorageContainer(parentCtx context.Context) error {
+	// Get a client to Azure Blob Storage
+	client, err := aeh.createStorageClient()
+	if err != nil {
+		return err
+	}
+
+	// Create the container
+	// This will return an error if it already exists
+	ctx, cancel := context.WithTimeout(parentCtx, resourceCreationTimeout)
+	defer cancel()
+	_, err = client.CreateContainer(ctx, aeh.metadata.StorageContainerName, &container.CreateOptions{
+		// Default is private
+		Access: nil,
+	})
+	if err != nil {
+		// Check if it's an Azure Storage error
+		resErr := &azcore.ResponseError{}
+		// If the container already exists, return no error
+		if errors.As(err, &resErr) && (resErr.ErrorCode == "ContainerAlreadyExists" || resErr.ErrorCode == "ResourceAlreadyExists") {
```

Comment:
create constants. 
Can we use any lib constants for this? As if the string changes, then this path is affected.

Reply:
I looked but sadly the constants are in the library but are not exported, they're in an internal package only (no idea why).

However these constants are defined in the Azure REST API specs so they will not change.